kratos/input_output/cad_json_input.h

```diff
@@ -0,0 +1,823 @@
+            << "[[x, y, z, weight]] or [id, [x, y, z, weight]]" << std::endl;
+
+        Vector cp = rParameters[number_of_entries - 1].GetVector();
+
+        return Kratos::make_shared<Point>(cp[0], cp[1], cp[2]);
+    }
+
+    ///@}
+    ///@name Utility functions
+    ///@{
+
+    /// Sets the geometry Id with either the 'brep_id' or the 'brep_name'.
+    template<class TGeometry>
+    static void SetIdOrName(
+        const Parameters& rParameters,
+        typename TGeometry::Pointer pGeometry)
+    {
+        if (rParameters.Has("brep_id")) {
+            pGeometry->SetId(rParameters["brep_id"].GetInt());
+        }
+        else if (rParameters.Has("brep_name")) {
+            pGeometry->SetId(rParameters["brep_name"].GetString());
+        }
+    }
+
+    /// Returns the string of either the 'brep_id' or the 'brep_name'. Used for output massages.
+    static std::string GetIdOrName(
+        const Parameters& rParameters)
+    {
+        if (rParameters.Has("brep_id")) {
+            return std::to_string(rParameters["brep_id"].GetInt());
+        }
+        else if (rParameters.Has("brep_name")) {
+            return rParameters["brep_name"].GetString();
+        }
+        else {
+            return "no_id_assigned";
+        }
+    }
+
+    /// Checks if one of the 'brep_id' or the 'brep_name' is provided.
+    static bool HasIdOrName(
+        const Parameters& rParameters)
+    {
+        return (rParameters.Has("brep_id") || rParameters.Has("brep_name"));
+    }
+
+    /// Returns the geometry with either the 'brep_id' or the 'brep_name'.
+    static typename GeometryType::Pointer GetGeometry(
+        const Parameters& rParameters,
+        ModelPart& rModelPart)
+    {
+        if (rParameters.Has("brep_id")) {
+            return rModelPart.pGetGeometry(rParameters["brep_id"].GetInt());
+        }
+        else { // if (rParameters["topology"][i].Has("brep_name"))
+            return rModelPart.pGetGeometry(rParameters["brep_name"].GetString());
+        }
+    }
+
+    /// Reads in a json formatted file and returns its KratosParameters instance.
+    static Parameters ReadParamatersFile(
+        const std::string& rDataFileName,
+        SizeType EchoLevel = 0)
+    {
+        // Check if rDataFileName ends with ".cad.json" and add it if needed.
+        std::string data_file_name = rDataFileName;
+        if (rDataFileName.compare(rDataFileName.size() - 9, 9, ".cad.json") != 0) {
+            data_file_name += ".cad.json";
+        }
+
+        KRATOS_INFO_IF("ReadParamatersFile", EchoLevel > 3)
+            << "Reading file: \"" << data_file_name << "\"" << std::endl;
+
+        std::ifstream cad_json_file;
+        cad_json_file.open(data_file_name);
+
+        // Convert std::ifstream into std::string
+        std::string cad_json_string((std::istreambuf_iterator<char>(cad_json_file)),
+            std::istreambuf_iterator<char>());
+
+        cad_json_file.close();

```

Comment:
I think this can be simplified, check this:

https://github.com/KratosMultiphysics/Kratos/blob/03cedee029ddf80924755609744926f0c32a56bb/kratos/utilities/read_materials_utility.cpp#L71-L76

Reply:
Nice. Thx. Changed it